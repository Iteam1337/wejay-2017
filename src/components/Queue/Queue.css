.Queue {
  height: 60vh;
  overflow: auto; }

.Queue > .Track:not(:last-child) {
  margin-bottom: 30px; }
