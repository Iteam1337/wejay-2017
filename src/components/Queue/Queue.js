// @flow

import './Queue.css'
import React from 'react'
import type { Track } from 'models/models'
import { timeParser } from 'utils/parsers'
import Cover from 'components/Cover'

type QueueProps = {
  tracks: Track[]
}

const Queue = ({ tracks }: QueueProps) => {
  if (!tracks) {
    return null
  }

  return (
    <ul>
      {tracks.map(track => (
        <li key={track.spotifyUri}>
          <Cover album={track.album} width={50} />
          {track.artists.map(t => t.name).join(',')} - {track.name} ({timeParser(track.duration)})
        </li>
      ))}
    </ul>
  )
}

export default Queue
