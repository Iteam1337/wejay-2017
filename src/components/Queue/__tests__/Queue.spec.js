import React from 'react'
import { shallow } from 'enzyme'
import Queue from '../Queue'

describe('components/Queue', () => {
  let component

  beforeEach(() => {
    component = shallow(
      <Queue
        tracks={[
          {
            name: 'Circles',
            spotifyUri: 1337
          }
        ]}
      />
    )
  })

  it('can render', () => {
    expect(component).toMatchSnapshot()
  })

  it('should display empty state when no queue', () => {
    component.setProps({
      tracks: []
    })

    expect(component.find('.EmptyState')).toMatchSnapshot()
  })
})
