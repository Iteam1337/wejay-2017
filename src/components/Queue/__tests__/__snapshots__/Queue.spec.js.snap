// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Queue can render 1`] = `
<ul>
  <li>
    Circles
  </li>
</ul>
`;
